File: python/src/Backstage.py
from src.Item import Item
from src.updateable import updateable

class Backstage(Item, updateable):

    def __init__(self, name, sell_in, quality):
        self.name = name
        self.sell_in = sell_in
        self.quality = quality

    def getSell_in(self):
        return self.sell_in
    
    def getQuality(self):
        return self.quality
    
    def updateQuality(self):
        if self.sell_in > 10:
            self.quality += 1
        elif self.sell_in <= 5:
            self.quality += 3
        else:
             self.quality += 2
        if self.sell_in <= 0:
            self.quality = 0
        self.maxQuality()
        self.sell_in -= 1

    def maxQuality(self):
        if self.quality >= 50:
            self.quality = 50

File: java/src/main/java/edu/poniperro/gildedrose/Backstage.java
package edu.poniperro.gildedrose;

public class Backstage extends Item implements Updateable {

    public Backstage(String name, int sell_in, int quality) {
        super(name, sell_in, quality);
    }

    @Override
    public void updateQuality() {
        if (getSell_in() > 10) {
            setQuality(getQuality() + 1);
        } else if (getSell_in() <= 5) {
            setQuality(getQuality() + 3);
        } else {
            setQuality(getQuality() + 2);
        }
        if (getSell_in() <= 0) {
            setQuality(0);
        }
        setSell_in();
        checkQuality();
    }

    private void checkQuality() {
        if (getQuality() >= 50) {
            setQuality(50);
        }
        if (getQuality() <= 0) {
            setQuality(0);
        }
    }
}
